sdk/src/types/block/core/block.rs

```diff
@@ -111,60 +115,29 @@ impl BlockHeader {
 
 impl WorkScore for BlockHeader {}
 
-impl Packable for BlockHeader {
-    type UnpackError = Error;
-    type UnpackVisitor = ProtocolParameters;
-
-    fn pack<P: Packer>(&self, packer: &mut P) -> Result<(), P::Error> {
-        self.protocol_version.pack(packer)?;
-        self.network_id.pack(packer)?;
-        self.issuing_time.pack(packer)?;
-        self.slot_commitment_id.pack(packer)?;
-        self.latest_finalized_slot.pack(packer)?;
-        self.issuer_id.pack(packer)?;
-
-        Ok(())
+fn verify_protocol_version<const VERIFY: bool>(
+    protocol_version: &u8,
+    params: &ProtocolParameters,
+) -> Result<(), Error> {
+    if VERIFY && *protocol_version != params.version() {
+        return Err(Error::ProtocolVersionMismatch {
+            expected: params.version(),
+            actual: *protocol_version,
+        });
     }
 
-    fn unpack<U: Unpacker, const VERIFY: bool>(
-        unpacker: &mut U,
-        protocol_params: &Self::UnpackVisitor,
-    ) -> Result<Self, UnpackError<Self::UnpackError, U::Error>> {
-        let protocol_version = u8::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
-
-        if VERIFY && protocol_version != protocol_params.version() {
-            return Err(UnpackError::Packable(Error::ProtocolVersionMismatch {
-                expected: protocol_params.version(),
-                actual: protocol_version,
-            }));
-        }
-
-        let network_id = u64::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
-
-        if VERIFY && network_id != protocol_params.network_id() {
-            return Err(UnpackError::Packable(Error::NetworkIdMismatch {
-                expected: protocol_params.network_id(),
-                actual: network_id,
-            }));
-        }
-
-        let issuing_time = u64::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
-
-        let slot_commitment_id = SlotCommitmentId::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
-
-        let latest_finalized_slot = SlotIndex::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
-
-        let issuer_id = AccountId::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+    Ok(())
+}
 
-        Ok(Self {
-            protocol_version,
-            network_id,
-            issuing_time,
-            slot_commitment_id,
-            latest_finalized_slot,
-            issuer_id,
-        })
+fn verify_network_id<const VERIFY: bool>(network_id: &u64, params: &ProtocolParameters) -> Result<(), Error> {
+    if VERIFY && *network_id != params.network_id() {
```

Comment:
```suggestion
    if VERIFY && network_id != &params.network_id() {
```
not that it mattered, but I would prefer this.

Reply:
We mostly dereference in verify functions so that would be inconsistent

Reply:
hmm, okay